### ISSUE TITLE
Old login resyncs all data

### ISSUE THREAD

**Issue by dr-waterstorm** [nextcloud/desktop#2523]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Expected behaviour
Updating an old installation (mid ~2019) and syncing files should work as expected. Meaning it should reflect the current state of the server and remove files which have been moved or removed on the server on the updated client as well.

### Actual behaviour
It re-uploaded all data that the server did not have. Instead of reflecting the changes on the server, it simply downloaded all new files, but also uploaded all of the old files. Which means the files which were moved are there twice now and the files which were removed are back on the server

### Steps to reproduce
1. Update any old installation to the current one
2. Login again
3. Let the client sync

### Client configuration
Client version: (latest) 3.0.2 Windows

Operating system: Windows

OS language: English

Qt version used by client package (Linux only, see also Settings dialog):

Client package (From Nextcloud or distro) (Linux only):

Installation path of client:
Default install

### Server configuration

Nextcloud version: 19.0.3

Storage backend (external storage): Standard HDD

**Comment by jospoortvliet** [comment]
Could you share a bit more info about what version(s) you went to-and-from? Both for server and client? I guess you don't have logs of what happened... If you do, that'd be great.

**Comment by dr-waterstorm** [comment]
Sorry for the delay, somehow I did not get a notification from Github.

I only upgraded the client, not the server. The server version was 19.0.3. 
I sadly do not recall the installed client version before the update. It was from mid 2019, when I last used the computer in question, so I guess it was around version 2.5.3. It told me to upgrade, which I did to 3.0.2 and this resulted in the described problem.

Sadly I also do not have the logs :(. But I wanted to post this anyways, in case anyone else tries to do this or even better: maybe the issue can be found somehow knowing what happened. 
This was a major issue as I had to spend a few hours to sort out the files from my backup.
